io/zenoh-transport/src/unicast/link.rs

```diff
@@ -391,3 +470,63 @@ async fn rx_task(
         rx_task_dgram(link, transport, lease, signal, rx_buffer_size).await
     }
 }
+
+#[cfg(feature = "transport_compression")]
+fn tx_compressed(
+    compression_enabled: bool,
+    bytes: &[u8],
+    compression_aux_buff: &mut Box<[u8]>,
+    link: &LinkUnicast,
+) -> ZResult<Vec<u8>> {
+    let mut buff = vec![];
+    if compression_enabled {
+        let zenoh_compress = ZenohCompress::default();
+        let compression_size = zenoh_compress
+            .write(&mut buff.writer(), (&bytes, compression_aux_buff))
```

Comment:
It's not clear to me why the compression_aux_buff is passed as argument and not its writer?